File: src/main/java/io/github/huacnlee/autocorrectIdeaPlugin/AutoCorrectActionOnSave.java
package io.github.huacnlee.autocorrectIdeaPlugin;

import com.intellij.ide.actionsOnSave.impl.ActionsOnSaveFileDocumentManagerListener;
import com.intellij.openapi.editor.Document;
import com.intellij.openapi.fileEditor.FileDocumentManager;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

public class AutoCorrectActionOnSave extends ActionsOnSaveFileDocumentManagerListener.ActionOnSave {
    @Override
    public boolean isEnabledForProject(@NotNull Project project) {
        return AppSettingsState.getInstance().enableFormatOnSave;
    }

    @Override
    public void processDocuments(@NotNull Project project, @NotNull Document @NotNull [] documents) {
        if (!AppSettingsState.getInstance().enableFormatOnSave) {
            return;
        }

        for (Document doc : documents) {
            VirtualFile file = FileDocumentManager.getInstance().getFile(doc);
            if (file == null) {
                return;
            }

            AutoCorrectExecutor.format(project, doc);
        }
    }
}

File: src/main/java/io/github/huacnlee/autocorrectIdeaPlugin/AutoCorrectActionOnSave.kt
package io.github.huacnlee.autocorrectIdeaPlugin

import com.intellij.ide.actionsOnSave.impl.ActionsOnSaveFileDocumentManagerListener.ActionOnSave
import com.intellij.openapi.editor.Document
import com.intellij.openapi.fileEditor.FileDocumentManager
import com.intellij.openapi.project.Project
import io.github.huacnlee.autocorrectIdeaPlugin.Formatter.format

class AutoCorrectActionOnSave : ActionOnSave() {
    override fun isEnabledForProject(project: Project): Boolean {
        return AppSettingsState.instance.enableFormatOnSave
    }

    override fun processDocuments(project: Project, documents: Array<Document>) {
        if (!AppSettingsState.instance.enableFormatOnSave) {
            return
        }
        for (doc in documents) {
            val file = FileDocumentManager.getInstance().getFile(doc) ?: return
            format(project, doc)
        }
    }
}